api/admin/controllers/file-storage.js

```diff
@@ -0,0 +1,32 @@
+const EventCreator = require('../../services/EventCreator');
+const { canAdminCreateSiteFileStorage } = require('../../authorizers/file-storage');
+const { serializeFileStorageService } = require('../../serializers/file-storage');
+const { wrapHandlers } = require('../../utils');
+const { Event } = require('../../models');
+const { adminCreateSiteFileStorage } = require('../../services/file-storage');
+
+module.exports = wrapHandlers({
+  async createSiteFileStorage(req, res) {
+    const { params, user } = req;
+
+    const siteId = parseInt(params.id, 10);
+
+    try {
+      const { site } = await canAdminCreateSiteFileStorage({ id: siteId });
+
+      const fss = await adminCreateSiteFileStorage(site);
+
+      EventCreator.audit(
+        Event.labels.ADMIN,
+        user,
+        'Site File Storage Service Created',
+        fss,
+      );
+
+      const data = serializeFileStorageService(fss);
+      return res.send(data);
+    } catch (error) {
+      return res.status(error.status).send(error);
```

Comment:
One issue with this pattern is that unhandled errors won't be logged anywhere, it will first show that it doesn't have an associated status:
`TypeError: Invalid status code: undefined. Status code must be an intege`

Reply:
Updated